jbmc/src/java_bytecode/remove_exceptions.cpp

```diff
@@ -589,6 +607,7 @@ void remove_exceptions(
     };
   remove_exceptionst remove_exceptions(
```

Comment:
:bulb: you might want to precondition class_hierachy if `type == remove_exceptions_typest::REMOVE_ADDED_INSTANCEOF` so that the caller gets a clearer error message.

Reply:
Kept the check in the constructor where it's not duplicated, but added explanatory text.